plugins/cookieauth.js

```diff
@@ -24,7 +24,7 @@ const CookieTokenManager = require('../lib/tokens/CookieTokenManager');
  * Cookie Authentication plugin.
  */
 class CookiePlugin extends BasePlugin {
-  constructor(client, cfg) {
+  constructor(client, cfg, reqDefaults) {
```

Comment:
I think this constructor can remain the same and we do either:
client.defaults() to get the requestDefaults to pass through to the TokenManager or just pass the client through, and call `defaults()` on it in the  TokenManager` - assuming that isn't the cause of the memory issue.